### ISSUE TITLE
moto should not take over `responses` defaults

### ISSUE THREAD

**Issue by rbarrois** [spulec/moto#3264]
We are using moto to mock AWS services (namely S3); and responses to control our access to third party APIs.

However, moto unconditionnally configures a global pass-through in `responses`:
At import time, `moto` configures responses:
https://github.com/spulec/moto/blob/master/moto/core/models.py#L249-L252

Once moto has been imported, it is no longer possible to use `responses` in one's code.

Since moto uses `@mock_XXX` idioms, it should only apply its configuration on code that is wrapped by such decorators/context managers.

**Comment by bblommers** [comment]
Agreed @rbarrois, thanks for raising this. Will mark it as an enhancement

**Comment by spulec** [comment]
I agree with this. I don't know that it will be possible with the current `responses` API though. I think this would require us to add some features there to be able to remove pass-throughs and such.

**Comment by escaped** [comment]
I think we should at least document this behaviour somewhere and suggest a way to remove the hook. I currently use a hack way to remove the hook globally.

```python
import responses

responses._default_mock.add_passthru  = tuple()
responses..add_passthru = responses._default_mock.add_passthru
```

This should be done where `moto` is imported, most likely this would be in the `conftest.py`.

...but this solution is not optimal (imo). Maybe we should turn the logic around, so you have to explicitly enable the passthru if you really need this as most test suites should work without it anyway. With this the documentation would state, that this project uses responses, which has the side effect that all external requests are blocked and if external requests are needed they can simply be enabled by explicitly using `responses.add_passthru(SOME_URL_PREFIX)`.

**Comment by andersk** [comment]
Can we just create our own `responses.RequestsMock` object instead of interfering with `responses._default_mock`? This seems to work (#3529).

**Comment by RobbieClarken** [comment]
I encountered this with tests that use `responses` but not `moto`. The errors only appeared when the tests were run after a test that did use `moto`.

```
self = <responses.RequestsMock object at 0x108894460>
adapter = <requests.adapters.HTTPAdapter object at 0x108894d60>, request = <PreparedRequest [GET]>
kwargs = {'cert': None, 'proxies': OrderedDict(), 'stream': False, 'timeout': None, ...}

    def _on_request(self, adapter, request, **kwargs):
>       match, match_failed_reasons = self._find_match(request)
E       TypeError: cannot unpack non-iterable Response object
```

The fix in #3529 looks sensible. In the meantime, this workaround works for me:

```python
@pytest.fixture
def configure_ssm():
    original_responses_mock = responses._default_mock
    responses._default_mock = responses.RequestsMock(assert_all_requests_are_fired=False)
    with mock_ssm():
        # ssm setup goes here
        yield
    responses._default_mock = original_responses_mock
```

**Comment by bblommers** [comment]
This should now be fixed as of moto >= 1.3.16.dev202
